### PR TITLE
Add library files

### PR DESCRIPTION
**Pull request by ajeetdsouza**

### CODE REVIEW

**Comment by shahidhk** [comment]
@ajeetdsouza Can you run the Haskell files through stylish-haskell?

Just do `stylish-haskell -i <file>`

File: src/CI.hs [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module CI ( Types.CI(..), isCI, getCI ) where
```
Comment (rakeshkky) [inline src/CI.hs]: exports in new line
```haskell
module CI
    ( Types.CI (..)
    , ----
    , -----
    ) where
```

File: src/CI.hs [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+import           Control.Arrow       ( (***) )
+
+import           Data.Bool           ( bool )
+import           Data.Foldable       ( find )
+import qualified Data.HashMap.Strict as HashMap
+import           Data.Maybe          ( isJust )
+import qualified Data.Text           as T
+
+import           System.Environment  ( getEnvironment )
```
Comment (rakeshkky) [inline src/CI.hs]: follow import style mentioned [here](https://kowainik.github.io/posts/2019-02-06-style-guide#imports)

File: src/CI/Types.hs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+instance Aeson.FromJSON VendorEnv where
+    parseJSON val = case val of
+        Aeson.String _ -> VendorEnvString <$> Aeson.parseJSON val
+        Aeson.Array _ -> VendorEnvList <$> Aeson.parseJSON val
+        Aeson.Object _ -> VendorEnvObject <$> Aeson.parseJSON val
+        _ -> fail "expected String, List[String], or Map[String, String] in vendor env"
+
+instance Aeson.ToJSON VendorEnv where
+    toJSON val = case val of
+        VendorEnvString key -> Aeson.toJSON key
```
Comment (rakeshkky) [inline src/CI/Types.hs]: `key` -> `name`

**Review by rakeshkky — COMMENTED** [review]
LGTM

**Review by shahidhk — APPROVED** [review]

**Pull request merged**